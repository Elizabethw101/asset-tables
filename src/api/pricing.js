import { prices } from './mockData/prices'

export const getPricing = () => {
    return prices;
}