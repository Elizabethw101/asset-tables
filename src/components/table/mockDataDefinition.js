
 export const mockDataDefinition = [
        {
          label: "Ticker",
          dataType: "text",
          sortType: "alpha",
          dataField: "ticker"
        },
        {
          label: "Price",
          dataType: "currency",
          sortType: "number",
          dataField: "price"
        },
        {
          label: "Asset Class",
          dataType: "asset",
          sortType: "assetClass",
          dataField: "assetClass"
        }
      ];
