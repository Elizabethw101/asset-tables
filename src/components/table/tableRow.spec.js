/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import TableRow from "./tableRow";
import { getPricing } from "../../api/pricing";
import {mockDataDefinition} from './mockDataDefinition';

describe("it tests the table row component", () => {
  let container = null;
  const pricing = getPricing();

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    container = null;
  });

  it("loads and renders the table row", () => {
    container = renderer.create(
        <TableRow
        rowData = {pricing[0]}
        key={1}
        colDefinition = {mockDataDefinition}
        /> 
    );
    expect(container.toJSON()).toMatchSnapshot();
  });
});
