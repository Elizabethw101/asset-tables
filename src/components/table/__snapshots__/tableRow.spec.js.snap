// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the table row component loads and renders the table row 1`] = `
<tr
  className="cell credit"
  scope="row"
>
  <td
    className="cell"
  >
    ALPHA
     
  </td>
  <td
    className="cell positive"
  >
    3150.67
     
  </td>
  <td
    className="cell"
  >
    Credit
     
  </td>
</tr>
`;
